File: Frontend/src/models/post-model.ts
export interface PostModel {
  id: number;
  text: string;
  authorId: number;
  authorFirstName: string;
  authorLastName: string;
  authorUsername: string;
  reactionCount: number;
  replyCount: number;
  reactionByUser: string;
}

export interface ReplyModel {
  postId: number;
  userId: string;
  replyText: string;
  replierFirstName?: string;
  replierLastName?: string;
  replierUsername: string;
}

File: Backend/models/post-model.go
package models

type Post struct {
	Id              int    `json:"id"`
	Text            string `json:"text"`
	AuthorId        int    `json:"authorId"`
	AuthorUsername  string `json:"authorUsername"`
	AuthorFirstName string `json:"authorFirstName"`
	AuthorLastName  string `json:"authorLastName"`
	ReplyCount      int    `json:"replyCount"`
	ReactionCount   int    `json:"reactionCount"`
	ReactionByUser  string `json:"reactionByUser"`
}
